### DISCUSSION TITLE
Selecting primitive types

### DISCUSSION THREAD

**Discussion by dplaskon** [maxtoroq/DbExtensions#14] [Q&A] [answered]
Perhaps I'm missing something obvious - but is there no way to get a result set of primitive types back without resorting to using a data reader.

For example, how do I do something along the lines of:

``` c#
var query = SQL
   .SELECT("p.ProductName")
   .FROM("Products p");

IEnumerable<string> names = conn.Map<string>(query);
```

**Comment by maxtoroq** [comment, marked as answer]
Currently, you have to explicitly provide a mapping delegate, otherwise it asumes you want to map columns to properties of the type parameter (String on this case):

``` csharp
IEnumerable<string> names = conn.Map<string>(query, r => r.GetString(0));
```

**Comment by maxtoroq** [comment]
With type inference you can ommit the type parameter:

``` csharp
IEnumerable<string> names = conn.Map(query, r => r.GetString(0));
```

**Comment by dplaskon** [comment]
Works perfectly, thanks again!
